### ISSUE TITLE
Possibly support custom media types?

### ISSUE THREAD

**Issue by Weffe** [Weffe/axios-api-versioning#4]
Reference: https://github.com/Microsoft/aspnet-api-versioning/wiki/Versioning-by-Media-Type#custom-media-types

Should we support custom media types? 

Currently, the mediaTypeKeyName and apiVersion is just appended to the Accept Header. The only way I can see this as a possibility is by providing a hook like `mediaTypeFormatter(acceptHeader: string): string`. If a user provides their own `mediaTypeFormatter` then we would use that instead of setting it ourselves like in the current way.

**Comment by Weffe** [comment]
Thought over this and came to the conclusion to green light this feature request. I think it's worth implementing to cover the edge case of custom media types. It should follow the same story like `mediaTypeKeyName` where it can be a prop thats passed to the `AxiosRequestConfig` or to the `IWithVersioningConfig`.

It can be like so:
```typescript
type mediaTypeFormatterFn = ({acceptHeader: string, apiVersion: string, mediaTypeKeyName: string}) => string;
```

With this in place, users who use mediaTypeFormatterFn are in full control of the Accept Header value. They can choose where to place the api version media type (e.g. at the beginning of the string) or even omit it all together. But they must return a string because we will use the return value as the Accept Header value.

**Comment by Weffe** [comment]
Done.

Example usage looks like this:

```typescript
import axios from 'axios';
import { withVersioning, VersioningStrategy, MediaTypeFormatterFn } from 'axios-api-versioning';

const formatter: MediaTypeFormatterFn = ({ apiVersion, mediaTypeKeyName, acceptHeader }) => (
    `fake-type/text, ${acceptHeader};${mediaTypeKeyName}=${apiVersion}`
)

const client = withVersioning(axios, {
    apiVersion: '2.2.3',
    versioningStrategy: VersioningStrategy.MediaType,
    mediaTypeFormatter: formatter
});
```

Ended up not letting users override the formatter in the `AxiosRequestConfig`. Thought that was too much flexibility and if you need to format the Accept Header differently then just create a new Axios instance with versioning and apply the formatter there in the config. Then just use the new instance configured with your formatter in your app.

I'll push soon so look for `v1.1.0`

**Comment by Weffe** [comment]
Changes merged in commit 7827b58428fb94f540c2e838f3452197b6bf999d
